pkg/k8s/controllers/reconciler.go

```diff
@@ -124,6 +137,11 @@ func (r *Reconciler) endReconcile(
 		return
 	}
 
+	if rr.Deleted {
```

Comment:
I think we should differentiate between "Not Found" or "other error", Not found could mean deleted.

Reply:
see below - r.Deleted is set ONLY on Not Found

Reply:
I also have a test for it - one for Not Found and another for 'any other err'

Reply:
I can rename to NotFound ...

Reply:
renamed Deleted to NotFound - although technically there is no case in which k8s would call Reconcile on never existing CRs :)